### ISSUE TITLE
Create 1-1 mapping for root-level config and transport config

### ISSUE THREAD

**Issue by prestonvasquez** [alpstable/gidari#265]
Right now in gidari.go we do this:

```go
// Config is the configuration object used to make programatic Transport requests.
type Config struct {
	transport.Config
}
```

This doesn't work for inline use in external packages since it relies on data from our the "internal/" packages. We don't want to expose data from the internal package to be used by third parties, limiting the number of things an end user can do.

Instead, we should keep things simple and just maintain a 1-1 copy of the transport data in the root-level gidari.go file. For example, the gidari.go file should now have

```go
type Authentication {
    // Needs to be filled in
}

type RateLimitConfig {
    // Needs to be filled in
}

type Request {
    // Needs to be filled in
}

type Config struct {
	RawURL            string    
	Authentication    Authentication 
	ConnectionStrings []string 
	Requests          []*Request 
	RateLimitConfig   *RateLimitConfig
	Logger            *logrus.Logger
	Truncate          bool
	URL *url.URL
}

mergeConfig(cfg *Config) *transport.Config {
   // Create a transport configuration out of the root-level configuration
}

// Transport will construct the transport operation using a "transport.Config" object.
func Transport(ctx context.Context, cfg *Config) error {
        tcfg := mergeConfig(cfg)
	if err := transport.Upsert(ctx, &tcfg); err != nil {
		return fmt.Errorf("unable to upsert the config: %w", err)
	}

	return nil
}
```

**Comment by mpeiffer** [comment]
Can I be assigned to this issue?

**Comment by prestonvasquez** [comment]
@mpeiffer Apologies for the late response, this has been closed by PR #288

**Comment by mpeiffer** [comment]
No worries!
